x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/percentiles_agg/config.ts

```diff
@@ -71,13 +88,36 @@ export function getPercentilesAggConfig(
 
       return {
         field: this.field as string,
-        percents: parsePercentsInput(this.aggConfig.percents),
+        percents: this.aggConfig.percents ?? [],
       };
     },
     isValid() {
-      return (
-        typeof this.aggConfig.percents === 'string' && isValidPercentsInput(this.aggConfig.percents)
-      );
+      const validationResult = validatePercentsInput(this.aggConfig);
+      this.aggConfig.errors = validationResult.errors;
+      return validationResult.isValid;
+    },
+    getErrorMessages() {
+      if (!this.aggConfig.errors?.length) return;
+
+      return this.aggConfig.errors.map((error) => ERROR_MESSAGES[error]);
     },
   };
 }
+
+const ERROR_MESSAGES: Record<ValidationResultErrorType, string> = {
+  INVALID_FORMAT: i18n.translate('xpack.transform.agg.popoverForm.invalidFormatError', {
+    defaultMessage: 'Percentile must be a valid number',
+  }),
+  PERCENTILE_OUT_OF_RANGE: i18n.translate(
+    'xpack.transform.agg.popoverForm.percentileOutOfRangeError',
+    {
+      defaultMessage: 'Percentiles must be between 0 and 100',
+    }
+  ),
+  NUMBER_TOO_PRECISE: i18n.translate('xpack.transform.agg.popoverForm.numberTooPreciseError', {
+    defaultMessage: 'Value is too precise',
```

Comment:
Just a suggestion: specifying what "too precise" means and providing guidance on acceptable values would be helpful. For example:
"Value is too precise; use fewer decimal places."
(or something similar that clearly instructs the user on how to avoid entering overly precise values).

Reply:
Done in: [#09e4c3a](https://github.com/elastic/kibana/pull/197816/commits/09e4c3adfa1d146668e45fca679f13d99e6be6f5)